doc/hardware/pinctrl/index.rst

```diff
@@ -292,6 +292,13 @@ requires to have one node for each pin mapping and state, since in general,
 nodes can not be re-used for multiple states. This method is discouraged if
 autogeneration is not an option.
 
+.. note::
+
+   Because all Devicetree information is parsed into a C header, it is important
+   to make sure its size is kept to a minimum. For this reason it is of special
+   importance to prefix pre-generated nodes with ``/omit-if-no-ref/``. This
```

Comment:
Nit: Not sure about the wording "it is of special importance".

Reply:
rephrased, pls check